internal/data/generated_metrics.go

```diff
@@ -34,60 +34,63 @@ type ResourceMetricsSlice struct {
-//
-// es := NewResourceMetricsSlice(4)
-// for i := 0; i < es.Len(); i++ {
-//     e := es.Get(i)
-//     // Here should set all the values for e.
-// }
-func NewResourceMetricsSlice(len int) ResourceMetricsSlice {
-	if len == 0 {
-		orig := []*otlpmetrics.ResourceMetrics(nil)
-		return ResourceMetricsSlice{&orig}
-	}
-	// Slice for underlying orig.
-	origs := make([]otlpmetrics.ResourceMetrics, len)
-	// Slice for wrappers.
-	wrappers := make([]*otlpmetrics.ResourceMetrics, len)
-	for i := range origs {
-		wrappers[i] = &origs[i]
-	}
-	return ResourceMetricsSlice{&wrappers}
-}
-
 func newResourceMetricsSlice(orig *[]*otlpmetrics.ResourceMetrics) ResourceMetricsSlice {
 	return ResourceMetricsSlice{orig}
 }
 
+// NewResourceMetricsSlice creates a ResourceMetricsSlice with 0 elements.
+// Can use "Resize" to initialize with a given length.
+func NewResourceMetricsSlice() ResourceMetricsSlice {
+	orig := []*otlpmetrics.ResourceMetrics(nil)
+	return ResourceMetricsSlice{&orig}
+}
+
 // Len returns the number of elements in the slice.
 func (es ResourceMetricsSlice) Len() int {
 	return len(*es.orig)
 }
 
-// Get the element at the given index.
+// At returns the element at the given index.
 //
 // This function is used mostly for iterating over all the values in the slice:
 // for i := 0; i < es.Len(); i++ {
-//     e := es.Get(i)
+//     e := es.At(i)
 //     ... // Do something with the element
 // }
-func (es ResourceMetricsSlice) Get(ix int) ResourceMetrics {
+func (es ResourceMetricsSlice) At(ix int) ResourceMetrics {
 	return newResourceMetrics((*es.orig)[ix])
 }
 
-// Remove the element at the given index from the slice.
-// Elements after the removed one are shifted to fill the emptied space.
-// The length of the slice is reduced by one.
-func (es ResourceMetricsSlice) Remove(ix int) {
-	(*es.orig)[ix] = (*es.orig)[len(*es.orig)-1]
-	(*es.orig)[len(*es.orig)-1] = nil
-	*es.orig = (*es.orig)[:len(*es.orig)-1]
-}
-
-// Resize the slice. This operation is equivalent with slice[from:to].
-func (es ResourceMetricsSlice) Resize(from, to int) {
-	*es.orig = (*es.orig)[from:to]
+// Resize is an operation that resizes the slice:
+// 1. If newLen is 0 then the slice is replaced with a nil slice.
+// 2. If the newLen < len then equivalent with slice[0:newLen].
+// 3. If the newLen > len then (newLen - len) empty elements will be appended to the slice.
+//
+// Here is how a new ResourceMetricsSlice can be initialized:
+// es := NewResourceMetricsSlice()
+// es.Resize(4)
+// for i := 0; i < es.Len(); i++ {
+//     e := es.At(i)
+//     // Here should set all the values for e.
+// }
+func (es ResourceMetricsSlice) Resize(newLen int) {
+	if newLen == 0 {
+		(*es.orig) = []*otlpmetrics.ResourceMetrics(nil)
+		return
+	}
+	oldLen := len(*es.orig)
+	if newLen < oldLen {
+		(*es.orig) = (*es.orig)[0:newLen]
+		return
+	}
+	// Probably this logic can be optimized.
+	origs := make([]otlpmetrics.ResourceMetrics, newLen-oldLen)
```

Comment:
Maybe give it a slightly better name:
```suggestion
	extraOrigs := make([]otlpmetrics.ResourceMetrics, newLen-oldLen)
```

Reply:
Done.